app/models/concerns/client_data_type.rb

```diff
@@ -1,14 +1,14 @@
-module ProposalDelegate
+module ClientDataType
```

Comment:
Simply `ClientData` is misleading. It isn't client data. It's the basic association definitions and validation rules that ClientData-type model classes should implement. It's like an abstract class for client data models, but as a concern.

`Mixin` has precedent in `PurchaseCardMixin` and doesn't repeat anything -- there is no `mixin` keyword in Ruby afaik.